src/hd.rs

```diff
@@ -275,7 +275,7 @@ impl HDWallet {
                     initial_import_size,
                     *rescan,
                 )
-                .with_context(|e| format!("invalid xpub {}: {:?}", xpub, e))?,
+                .with_context(|| format!("invalid xpub {}", xpub))?,
```

Comment:
I'd look at the actual errors this produce to see that they look like you want them too

Reply:
I checked, looks good!